config/config.go

```diff
@@ -36,6 +36,10 @@ type Config struct {
 	// MainContainer holds container customization for the main container
 	MainContainer *apiv1.Container `json:"mainContainer,omitempty"`
 
+	// TransientErrorPattern specifies the pattern to match for errors that can be seen as transient
+	// and retryable.
+	TransientErrorPattern string `json:"transientErrorPattern,omitempty"`
```

Comment:
I think this should be a rarely used feature, so an environment variable would be the way to configure it - it is much simpler too.

Reply:
In our case, we'd like to whitelist non-critical errors that can be retried for issues during cluster maintenance/migration, etc., especially for non-production clusters where certain services may not be available.

Good suggestion. I'll add an env var to configure this instead.

Reply:
Just updated. Please take another look.